### PR TITLE
distinct subjects

### PR DESCRIPTION
**Pull request by camallen**
this will create a unique list of subject_id, priority, group based on the sms data. Not sure if cellect will take it into account when loading for non-grouped workflows though.

### CODE REVIEW

File: lib/cellect/server/adapters/panoptes.rb [hunk 1]
```diff
@@ -58,8 +58,9 @@ def load_data_for(workflow_id)
               .joins(:workflows)
               .where(workflows: {id: workflow_id})
               .joins("LEFT OUTER JOIN subject_workflow_counts ON subject_workflow_counts.subject_id = set_member_subjects.subject_id")
-              .where('subject_workflow_counts.id IS NULL OR subject_workflow_counts.retired_at IS NULL')
+              .where('subject_workflow_counts.retired_at IS NULL')
```
Comment (parrish) [inline lib/cellect/server/adapters/panoptes.rb]: I thought we needed the id check for non-classified subjects?

Comment (camallen) [inline lib/cellect/server/adapters/panoptes.rb]: if the left outer is empty the retired_at will be null as well as the id, right?

Comment (parrish) [inline lib/cellect/server/adapters/panoptes.rb]: Yep.  Good call.  Does `distinct` need to specify that it's distinct on the `subject_id`?

Comment (camallen) [inline lib/cellect/server/adapters/panoptes.rb]: the sql was showing distinct on the whole select clause.
```
SELECT DISTINCT set_member_subjects.subject_id, set_member_subjects.priority, set_member_subjects.subject_set_id
FROM set_member_subjects
INNER JOIN subject_sets ON subject_sets.id = set_member_subjects.subject_set_id
INNER JOIN subject_sets_workflows ON subject_sets_workflows.subject_set_id = subject_sets.id INNER JOIN workflows ON workflows.id = subject_sets_workflows.workflow_id LEFT OUTER JOIN subject_workflow_counts ON subject_workflow_counts.subject_id = set_member_subjects.subject_id
WHERE workflows.id = 41 AND (subject_workflow_counts.retired_at IS NULL)
```

**Comment by parrish** [comment]
:+1:

**Pull request merged**